packages/engine/src/physics/systems/PhysicsSystem.ts

```diff
@@ -116,20 +97,35 @@ export default async function PhysicsSystem(world: World) {
       world.physics.doRaycast(getComponent(entity, RaycastComponent))
     }
 
+    for (const entity of networkColliderQuery()) {
+      const network = getComponent(entity, NetworkObjectComponent)
+
+      // Set network state to physics body pose for objects not owned by this user.
+      if (network.ownerId === Engine.userId) continue
+
+      const collider = getComponent(entity, ColliderComponent)
+      const transform = getComponent(entity, TransformComponent)
+      const body = collider.body as PhysX.PxRigidDynamic
+
+      teleportRigidbody(body, transform.position, transform.rotation)
+    }
+
     for (const entity of colliderQuery()) {
+      // Hackish fix for now to disable some physics effects in editor
+      // We should prolly not register the physics system when loading editor
+      if (Engine.isEditor) return
```

Comment:
`return` will take us out of the entire ECS loop here, if we want to exit the for loop, use `continue`. as per the style guide, we also want to take all these loops out of the main system function into their own module scope functions as a more composable style of programming